developer/src/kmc-keyboard/src/compiler/keys.ts

```diff
@@ -71,15 +72,17 @@ export class KeysCompiler extends SectionCompiler {
 
     // Finally, kmap
     // Use LayerMap + keys to generate compiled keys for hardware
-    const theLayers = this.keyboard.layers?.[0]; // TODO-LDML: handle >1 layers. #8160
-
-    if(theLayers?.form == 'hardware') {
+    const hardwareLayers = this.keyboard.layers?.filter(({form}) => form !== 'touch');
```

Comment:
Given we are doing this twice -- here and on line 43 -- we should have a local helper function `this.hardwareLayers()`